### PR TITLE
Add typings for `focus-within`

### PR DESCRIPTION
**Pull request by eramdam**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/focus-within/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    attr?: boolean;
+    className?: string;
+}
+
+type FocusWithin = (document: HTMLDocument, opts?: FocusWithinOpts) => void;
+
+const focusWithin: FocusWithin;
+}
+
+export default focusWithin;
```
Comment (mhegazy) [inline types/focus-within/index.d.ts]: should be `export = focusWithin`. see https://github.com/DefinitelyTyped/DefinitelyTyped#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default for more details

File: types/focus-within/test/focus-within-import-default-tests.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/*
+ * This file tests the typings of the default export of `focus-within`
+ */
+import focusWithin from 'focus-within';
```
Comment (mhegazy) [inline types/focus-within/test/focus-within-import-default-tests.ts]: this should be `import focusWithin = require("focus-within");` to ensure it works in older versions of TS that the tester uses.

File: types/focus-within/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Type definitions for focus-within 1.0
+// Project: https://github.com/jonathantneal/focus-within#readme
+// Definitions by: Damien Erambert <https://github.com/eramdam>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare global {
```
Comment (mhegazy) [inline types/focus-within/index.d.ts]: instead, just use UMD pattern. in other words:

```ts
declare function focusWithin(document: HTMLDocument, opts?: FocusWithinOpts): void;
declare namespace focusWithin {
    interface FocusWithinOpts {
      attr?: boolean;
      className?: string;
    }
}

export as namespace focusWithin;
export = focusWithin;
```

**Comment by eramdam** [comment]
@mhegazy Thanks for reviewing so quickly! 😊 I updated my PR according to your comments

**Pull request merged**